app/src/main/java/com/duckduckgo/app/systemsearch/DeviceAppLookup.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.systemsearch
+
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.content.pm.PackageManager.GET_META_DATA
+import androidx.annotation.WorkerThread
+import kotlin.text.RegexOption.IGNORE_CASE
+
+data class DeviceApp(
+    val shortName: String,
+    val packageName: String,
+    val launchIntent: Intent
+)
+
+interface DeviceAppLookup {
+    @WorkerThread
+    fun query(query: String): List<DeviceApp>
+}
+
+class InstalledDeviceAppLookup(private val appListProvider: DeviceAppListProvider) : DeviceAppLookup {
+
+    private val apps by lazy { appListProvider.get() }
+
+    @WorkerThread
+    override fun query(query: String): List<DeviceApp> {
+        if (query.isBlank()) return emptyList()
+        val regex = ".*\\b${query}.*".toRegex(IGNORE_CASE)
```

Comment:
Can you explain this regex with a comment?

Reply:
How does a rename to `wordPrefixRegex` sound?